### PR TITLE
Initial upload

### PR DESCRIPTION
**Pull request by artivis**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,2 +1,103 @@
...
+
+## Notes
+
+Note that this snap is part of an integrated snaps deployment.
+Other recommended snaps to be installed are,
+
+- [micro-xrce-dds-agent](LINK)
+- [sllidar-ros2](https://snapcraft.io/sllidar-ros2)
+- [rosbot-xl](https://snapcraft.io/rosbot-xl)
+- [rosbot-xl-teleop](https://snapcraft.io/rosbot-xl-teleop)
```
Comment (Kotochleb) [inline README.md]:
```suggestion
> **Note**
> This snap is part of an integrated snaps deployment.
> 
> Other recommended snaps to be installed are,
> 
> - [micro-xrce-dds-agent](LINK)
> - [sllidar-ros2](https://snapcraft.io/sllidar-ros2)
> - [rosbot-xl-teleop](https://snapcraft.io/rosbot-xl-teleop)
> - [rosbot-xl-nav](https://snapcraft.io/rosbot-xl-nav)
```

File: snap/snapcraft.yaml [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+
+  reset-config:
+    command: usr/bin/reset_config.sh
+
+  slam:
+    command: usr/bin/slam_launcher.sh
+    daemon: simple
+    # @todo(artivis) figure out why can't we set those in the env
+    # environment:
+    #   LD_LIBRARY_PATH: "$SNAP/opt/ros/underlay_ws/usr/lib/$SNAPCRAFT_ARCH_TRIPLET/blas:$SNAP/opt/ros/underlay_ws/usr/lib/$SNAPCRAFT_ARCH_TRIPLET/lapack"
```
Comment (Kotochleb) [inline snap/snapcraft.yaml]: This can be already removed

File: snap/hooks/configure [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+	"0") ;;
+	"1") ;;
+	*)
+		>&2 echo "'$SIMULATION' is not a supported value for simulation." \
+      "Possible values are true, false, 0, 1"
+	    logger -t ${SNAP_NAME} "'$SIMULATION' is not a supported value for simulation." \
+      "Possible values are true, false, 0, 1"
+		return 1
+		;;
+esac
```
Comment (Kotochleb) [inline snap/hooks/configure]:
```suggestion
case "$SIMULATION" in
	"True") ;;
	"False") ;;
	*)
		>&2 echo "'$SIMULATION' is not a supported value for simulation." \
      "Possible values are True, False"
	    logger -t ${SNAP_NAME} "'$SIMULATION' is not a supported value for simulation." \
      "Possible values are True, False"
		return 1
		;;
esac
```

File: snap/snapcraft.yaml
Comment (Kotochleb) [inline snap/snapcraft.yaml:1]: Separate app to save the map would be a nice addition

File: snap/local/mux_select_nav_vel.sh [hunk 4]
```diff
@@ -0,0 +1,4 @@
+#!/usr/bin/bash
+
+$SNAP/opt/ros/underlay_ws/opt/ros/$ROS_DISTRO/bin/ros2 service call /mux/select topic_tools_interfaces/srv/MuxSelect "{topic: /nav_vel}"
```
Comment (Kotochleb) [inline snap/local/mux_select_nav_vel.sh]:
```suggestion
ros2 service call /mux/select topic_tools_interfaces/srv/MuxSelect "{topic: /nav_vel}"
```
As far as I am concerned this is no longer required

File: snap/local/map_server_launcher.sh
Comment (Kotochleb) [inline snap/local/map_server_launcher.sh:1]: This file is not used anywhere...

File: README.md [hunk 5]
```diff
@@ -1,2 +1,103 @@
...
+
+## Use
+
+### Slam
+
+This application allows for mapping the environment.
+It can be launch as follows,
+
+```bash
+snap start rosbot-xl-teleop.slam
```
Comment (Kotochleb) [inline README.md]:
```suggestion
snap start rosbot-xl-nav.slam
```

File: README.md [hunk 6]
```diff
@@ -1,2 +1,103 @@
...
+```
+
+One can then drive the robot around to create a 2D representation of the environment.
+Once the area is covered, stop the application with,
+
+```bash
+snap stop rosbot-xl-teleop.slam
```
Comment (Kotochleb) [inline README.md]:
```suggestion
snap stop rosbot-xl-nav.slam
```

File: README.md [hunk 7]
```diff
@@ -1,2 +1,103 @@
...
+
+### Localisation
+
+The localisation allows the robot, as its name suggests,
+to localise itself in the map created by the slam application.
+Note that it automatically uses the last map created by the slam application.
+It can be started and stopped respectively with,
+
+```bash
+snap start rosbot-xl-teleop.localisation
```
Comment (Kotochleb) [inline README.md:71]:
```suggestion
snap start rosbot-xl-nav.localisation
```

File: README.md [hunk 8]
```diff
@@ -1,2 +1,103 @@
...
+```
+
+```bash
+snap stop rosbot-xl-teleop.localisation
```
Comment (Kotochleb) [inline README.md:75]:
```suggestion
snap stop rosbot-xl-nav.localisation
```

File: README.md [hunk 9]
```diff
@@ -1,2 +1,103 @@
...
+you can simply issue the command `rosbot-xl-nav.reset-config` which will reset all configuration files.
+
+### Navigation
+
+The navigation application allows the robot to autonomously move around to a defined goal while avoiding obstacles.
+Note that it automatically uses the last map created by the slam application.
+It can be started and stopped respectively with,
+
+```bash
+rosbot-xl-teleop.navigation
```
Comment (Kotochleb) [inline README.md:89]:
```suggestion
rosbot-xl-nav.navigation
```

File: README.md [hunk 10]
```diff
@@ -1,2 +1,103 @@
...
+Upon installation, depending on your operating system,
+you may have to manually connect the snap interface.
+You can verify that with the following command,
+
+```bash
+$ snap connections rosbot-xl
+Interface            Plug                    Slot                            Notes
+content[ros-humble]  rosbot-xl:ros-humble    ros-humble-ros-base:ros-humble  manual
+network              rosbot-xl:network       :network                        -
+network-bind         rosbot-xl:network-bind  :network-bind                   -
```
Comment (Kotochleb) [inline README.md]:
```suggestion
$ snap connections rosbot-xl-nav
Interface            Plug                        Slot                            Notes
content[ros-humble]  rosbot-xl-nav:ros-humble    ros-humble-ros-base:ros-humble  manual
network              rosbot-xl-nav:network       :network                        -
network-bind         rosbot-xl-nav:network-bind  :network-bind                   -

```

File: README.md [hunk 11]
```diff
@@ -1,2 +1,103 @@
...
+```
+
+One can then drive the robot around to create a 2D representation of the environment.
+Once the area is covered, stop the application with,
+
+```bash
+snap stop rosbot-xl-teleop.slam
+```
+
+Upon stopping the application automatically save the map at `${SNAP_COMMON}/maps/current_map.{png,yaml}`.
```
Comment (Kotochleb) [inline README.md]:
```suggestion
Upon stopping the application automatically save the map at `$SNAP_COMMON/maps/current_map.{png,yaml}`.
```
Just to keep it coherent with all the rest

**Review by Kotochleb — CHANGES_REQUESTED** [review]
Everything works fine. Aside of that small changes it is ok

File: snap/snapcraft.yaml
Comment (artivis) [reply snap/snapcraft.yaml:1]: Let us iterate on that later ;)

**Review by Kotochleb — APPROVED** [review]
After all of the discussions, it looks and works good. I am closing this PR

File: README.md [hunk 12]
```diff
@@ -1,2 +1,106 @@
...
+
+## Use
+
+### Slam
+
+This application allows for mapping the environment.
+It can be launch as follows,
+
+```bash
+snap start rosbot-xl-nav.slam
```
Comment (Guillaumebeuzeboc) [inline README.md:47]: sudo is required to start/stop services

**Pull request merged**